mongoengine/base/datastructures.py

```diff
@@ -22,7 +22,7 @@ def __init__(self, dict_items, instance, name):
         EmbeddedDocument = _import_class('EmbeddedDocument')
 
         if isinstance(instance, (Document, EmbeddedDocument)):
-            self._instance = weakref.proxy(instance)
+            self._instance = instance
```

Comment:
Without this change we get a few errors like this in a handful of places in the platform (for example the syllabus page)

<img width="1106" alt="screen shot 2018-08-16 at 1 22 29 pm" src="https://user-images.githubusercontent.com/904782/44224108-82979900-a157-11e8-8eeb-ac6f944ce373.png">

To reproduce.  Complete a module and then return to the syllabus page.  It doesn't happen on every course or every module, but when it does it's because a variable on the _instance has also gone away.  I first discovered that I could fix it by simply assigning _instance to ANY other class variable.... Which makes sense because then it has a hard reference.  So the easiest way to 'fix' this is to simply undo the weakref...  

I consider this a dicey change.  I'm not sure WHY they used a weak reference here, but my suspicion was they were trying to build a pseudo garbage collector instead of relying on pythons garbage collector.  Which means there could be memory ramifications from undoing this.  Where I image this would be an issue is when we are holding LOTS of mongoengine objects in memory.   

I haven't been able to find a root cause as to why the reference goes away.  Unless others have good ideas I think we try this out and perf test the hell out of it to make sure there aren't memory leaks or spikes in normal memory usage because of this.

FYI This is in also BaseList, but they do a similar assignment in BaseList as well.  

For this PR I have removed them both, but I haven't actually seen an Error from the BaseList weakref.

Reply:
Interesting research, I wonder will we see memory decrease/increase after this

Reply:
thanks for the research - I'd be interested in doing some searching inside the mongoengine repository to sleuth out why weakref was used.

Reply:
It looks like they were added here: https://github.com/MongoEngine/mongoengine/commit/9e67941badd0ddd680b2fc582499f7c660b2ea4b partly in response to this issue: https://github.com/MongoEngine/mongoengine/issues/535. Still looking into what it means for this change.